### PR TITLE
[crashpad] fix linking on macOS

### PR DESCRIPTION
**Pull request by promag**
**Describe the pull request**

- #### What does your PR fix?
  Fixes linking on macOS

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Comment by promag** [comment]
Fix discussed here https://groups.google.com/a/chromium.org/g/crashpad-dev/c/XVggc7kvlNs

File: ports/crashpad/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "crashpad",
-  "version-date": "2022-04-16",
-  "port-version": 3,
+  "version-date": "2022-09-05",
+  "port-version": 4,
```
Comment (JonLiu1993) [inline ports/crashpad/vcpkg.json:4]:
```suggestion
```

Comment (JonLiu1993) [reply ports/crashpad/vcpkg.json:4]: @promag, Thanks for your pr, if you update a port, you should remove the `port-version`

File: versions/c-/crashpad.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "3465990a521f4cb143a5c5fbcb285cce50add96e",
+      "version-date": "2022-09-05",
+      "port-version": 4
+    },
+    {
+      "git-tree": "955373b979b70472dc4dee66726e2fae83c69e4c",
+      "version-date": "2022-04-16",
+      "port-version": 4
+    },
```
Comment (JonLiu1993) [inline versions/c-/crashpad.json:12]:
```suggestion
    {
      "git-tree": "955373b979b70472dc4dee66726e2fae83c69e4c",
      "version-date": "2022-04-16",
      "port-version": 4
    },
```

Comment (JonLiu1993) [reply versions/c-/crashpad.json:12]: Each time we commit pr, we can only add one "versions" record, which is beneficial for us to backtrack later versions

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by promag** [comment]
@JonLiu1993 apologies, removed `port-version` key from ports/crashpad/vcpkg.json.

**Review by JonLiu1993 — APPROVED** [review]

File: ports/crashpad/portfile.cmake [hunk 3]
```diff
@@ -113,6 +113,10 @@ install_headers("${SOURCE_PATH}/third_party/mini_chromium/mini_chromium/build")
 
 file(COPY "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/gen/build/chromeos_buildflags.h" DESTINATION "${CURRENT_PACKAGES_DIR}/include/${PORT}/build")
 file(COPY "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/gen/build/chromeos_buildflags.h.flags" DESTINATION "${CURRENT_PACKAGES_DIR}/include/${PORT}/build")
+if(APPLE)
+file(COPY "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/obj/util/libmig_output.a" DESTINATION "${CURRENT_PACKAGES_DIR}/debug/lib")
+file(COPY "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/obj/util/libmig_output.a" DESTINATION "${CURRENT_PACKAGES_DIR}/lib")
+endif()
```
Comment (JonLiu1993) [inline ports/crashpad/portfile.cmake:119]:
```suggestion
if(VCPKG_TARGET_IS_OSX)
    file(COPY "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/obj/util/libmig_output.a" DESTINATION "${CURRENT_PACKAGES_DIR}/debug/lib")
    file(COPY "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/obj/util/libmig_output.a" DESTINATION "${CURRENT_PACKAGES_DIR}/lib")
endif()
```

Comment (JonLiu1993) [reply ports/crashpad/portfile.cmake:119]: The representation and format here need to be modified

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/crashpad/portfile.cmake (hunk 3 above)
Comment (promag) [reply ports/crashpad/portfile.cmake]: Done.

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**